docs/ml-clustering.md

```diff
@@ -104,4 +104,38 @@ Refer to the [Java API docs](api/java/org/apache/spark/ml/clustering/LDA.html) f
 {% include_example java/org/apache/spark/examples/ml/JavaLDAExample.java %}
 </div>
 
-</div>
\ No newline at end of file
+</div>
+
+## Bisecting k-means
+
+
+Bisecting k-means is a kind of [hierarchical clustering](https://en.wikipedia.org/wiki/Hierarchical_clustering) in the fashion of
+division: all observations start in one cluster, and splits are performed recursively as one moves down the hierarchy.
+
+Bisecting K-means can often be much faster than regular K-means, but it will generally produce a different clustering.
+
```

Comment:
Since we intend to make the `ml` docs "complete" (as in, as complete as `mllib`), could we detail the parameters as is done in the doc for the `mllib` algorithm.

This will probably need to be done across the board (but the doc parity work will be covered as part of [SPARK-14815](https://issues.apache.org/jira/browse/SPARK-14815))